graph/iterator/resolver_test.go

```diff
@@ -0,0 +1,130 @@
+package iterator_test
+
+import (
+	"testing"
+	"github.com/cayleygraph/cayley"
+	"github.com/cayleygraph/cayley/quad"
+	"context"
+	"github.com/cayleygraph/cayley/graph"
+	"github.com/cayleygraph/cayley/graph/iterator"
+)
+
+func TestResolverIteratorIterate(t *testing.T) {
+	var ctx context.Context
+	qs, err := cayley.NewMemoryGraph()
+	if err != nil {
+		t.Fatalf("error creating graph: %v", err)
+	}
+	nodes := []quad.Value{
+		quad.String("1"),
+		quad.String("2"),
+		quad.String("3"),
+		quad.String("4"),
+		quad.String("5"),
+	}
+	expected := make(map[quad.Value]graph.Value)
+	for _, node := range nodes {
+		qs.AddQuad(quad.Make(quad.String("0"), nil, node, nil))
```

Comment:
This only works because it's an in-memory store. It's better to specify a predicate here.